### PR TITLE
Added "Follow Redirects" as a top level parameter to all request functions

### PR DESCRIPTION
**Pull request by owenkealey**
On multiple occassions, I have needed to specify "follow redirects" on/off in specific GET/POST requests that I don't feel warranted the boilerplate of implementing BaseRequest. Thus, I added followRedirects as a top level parameter to all core http functions(.get, .post, .head etc).

### CODE REVIEW

File: lib/src/response.dart [hunk 1]
```diff
@@ -25,7 +26,18 @@ class Response extends BaseResponse {
   /// [RFC 2616][].
   ///
   /// [RFC 2616]: http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html
-  String get body => _encodingForHeaders(headers).decode(bodyBytes);
+  String parseBody() {
```
Comment (kevmoo) [inline lib/src/response.dart:29]: This seems unrelated to the title of the PR

**Review by kevmoo — COMMENTED** [review]
Looks like some `build` files were added that should be removed!

File: lib/src/response.dart (hunk 1 above)
Comment (owenkealey) [reply lib/src/response.dart:29]: Your'e correct. I use my fork for personal projects and forgot this PR was still open. I am going to close and open a new one with just the follow redirects only.

Comment (kevmoo) [reply lib/src/response.dart:29]: I'd do seperate PRs for protli and proxy support
